analyzers/tests/SonarAnalyzer.Test/Rules/MessageTemplatesShouldBeCorrectTest.cs

```diff
@@ -0,0 +1,125 @@
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.Rules.CSharp;
+
+namespace SonarAnalyzer.Test.Rules;
+
+[TestClass]
+public class MessageTemplatesShouldBeCorrectTest
+{
+    private static readonly VerifierBuilder Builder = new VerifierBuilder<MessageTemplatesShouldBeCorrect>();
+
+    [TestMethod]
+    public void MessageTemplatesShouldBeCorrect_CS() =>
+        Builder
+        .AddReferences(NuGetMetadataReference.MicrosoftExtensionsLoggingAbstractions())
+        .AddPaths("MessageTemplatesShouldBeCorrect.cs")
+        .Verify();
+
+    [DataTestMethod]
+    [DataRow("LogCritical")]
+    [DataRow("LogDebug")]
+    [DataRow("LogError")]
+    [DataRow("LogInformation")]
+    [DataRow("LogTrace")]
+    [DataRow("LogWarning")]
+    public void MessageTemplatesShouldBeCorrect_MicrosoftExtensionsLogging_CS(string methodName) =>
+        Builder
+        .AddReferences(NuGetMetadataReference.MicrosoftExtensionsLoggingAbstractions())
+        .AddSnippet($$$"""
+            using System;
+            using Microsoft.Extensions.Logging;
+
+            public class Program
+            {
+                public void Method(ILogger logger, string user, int count)
+                {
+                    Console.WriteLine("Login failed for {User", user);                  // Compliant
+                    logger.{{{methodName}}}("Login failed for {User}", user);           // Compliant
+
+                    logger.{{{methodName}}}("{", user);                                 // Noncompliant
+                    logger.Log(LogLevel.Information, "{", user);                        // Noncompliant
+                    LoggerExtensions.{{{methodName}}}(logger, "{", user);               // Noncompliant
+                }
+            }
+            """).Verify();
+
+    [DataTestMethod]
+    [DataRow("Debug")]
+    [DataRow("Error")]
+    [DataRow("Information")]
+    [DataRow("Fatal")]
+    [DataRow("Warning")]
+    [DataRow("Verbose")]
+    public void MessageTemplatesShouldBeCorrect_Serilog_CS(string methodName) =>
+        Builder
+        .AddReferences(NuGetMetadataReference.Serilog(Constants.NuGetLatestVersion))
+        .AddSnippet($$$"""
+            using System;
+            using Serilog;
+            using Serilog.Events;
+
+            public class Program
+            {
+                public void Method(ILogger logger, string user, int count)
+                {
+                    Console.WriteLine("Login failed for {User", user);                  // Compliant
+                    logger.{{{methodName}}}("Login failed for {User}", user);           // Compliant
+
+                    logger.{{{methodName}}}("{", user);                                 // Noncompliant
+                    Log.{{{methodName}}}("{", user);                                    // Noncompliant
+                    Log.Write(LogEventLevel.Verbose, "{", user);                        // Noncompliant
```

Comment:
Same here. It should either be a separate test or use a LogEventLevel parameter.